Support of Stencil 2.0

Are were plans to support Stencil 2.0 and when would that be? We can't migrate our projects to Stencil 2.x since `stencil-sass` does not support it.

@claviska how are you testing this? It doesn't work at all:
```
$ npm update
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: frontend@0.0.1
npm ERR! Found: @stencil/core@2.2.0
npm ERR! node_modules/@stencil/core
npm ERR!   dev @stencil/core@">=2.2.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @stencil/core@"^1.0.2" from @stencil/sass@1.3.2
npm ERR! node_modules/@stencil/sass
npm ERR!   dev @stencil/sass@">=1.3.2" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/camdennarzt/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/camdennarzt/.npm/_logs/2020-10-28T19_35_10_781Z-debug.log
```

It's alive and well in two projects I maintain. The public one can be found here: https://github.com/shoelace-style/shoelace/blob/next/package.json#L43

> npm ERR! Fix the upstream dependency conflict, or retry
> npm ERR! this command with --force, or --legacy-peer-deps

It looks like `@stencil/core` is a peer dependency, and the way those get installed by npm changed in npm 7, so you probably need to install core manually:

```bash
npm install -D @stencil/core
```

Or use the `--legacy-peer-deps` flag as [discussed here](https://blog.npmjs.org/post/626173315965468672/npm-v7-series-beta-release-and-semver-major#peer-dependencies).

Here's what I get following those steps with node: `v15.0.1`:
```
camdennarzt@WALLE:/tmp $ npm init stencil
Need to install the following packages:
  create-stencil
Ok to proceed? (y) y

✔ Pick a starter › app
✔ Project name › blah

✔ All setup  in 20 ms

  $ npm start
    Starts the development server.

  $ npm run build
    Builds your components/app in production mode.

  $ npm test
    Starts the test runner.


  We suggest that you begin by typing:

   $ cd blah
   $ npm install
   $ npm start

  Further reading:

   - https://github.com/ionic-team/stencil-app-starter

  Happy coding! 🎈

camdennarzt@WALLE:/tmp $ cd blah/
camdennarzt@WALLE:/tmp/blah $ npm install

added 3 packages, and audited 3 packages in 1s

found 0 vulnerabilities
camdennarzt@WALLE:/tmp/blah $ cat package.json 
{
  "name": "blah",
  "private": true,
  "version": "0.0.1",
  "description": "Stencil App Starter",
  "scripts": {
    "build": "stencil build",
    "start": "stencil build --dev --watch --serve",
    "test": "stencil test --spec --e2e",
    "test.watch": "stencil test --spec --e2e --watchAll",
    "generate": "stencil generate"
  },
  "devDependencies": {
    "@stencil/core": "^2.0.0",
    "@stencil/router": "^1.0.1"
  },
  "license": "MIT"
}
camdennarzt@WALLE:/tmp/blah $ npm install -D @stencil/sass
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: blah@0.0.1
npm ERR! Found: @stencil/core@2.2.0
npm ERR! node_modules/@stencil/core
npm ERR!   dev @stencil/core@"^2.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @stencil/core@"^1.0.2" from @stencil/sass@1.3.2
npm ERR! node_modules/@stencil/sass
npm ERR!   dev @stencil/sass@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/camdennarzt/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/camdennarzt/.npm/_logs/2020-10-28T20_01_49_683Z-debug.log
```

--- a/package.json
+++ b/package.json
@@ -19,7 +19,7 @@
     "test.watch": "jest --watch"
   },
   "peerDependencies": {
-    "@stencil/core": "^1.0.2"
+    "@stencil/core": ">=1.0.2"
   },
   "devDependencies": {
     "@rollup/plugin-node-resolve": "^8.0.1",

